code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,354 @@
+#define TUMOR_INACTIVE 0
+#define TUMOR_ACTIVE 1
+#define TUMOR_PASSIVE 2
+
+//Elite mining mobs
+/mob/living/simple_animal/hostile/asteroid/elite
+	name = "elite"
+	desc = "An elite monster, found in one of the strange tumors on lavaland."
+	icon = 'icons/mob/lavaland/lavaland_elites.dmi'
+	faction = list("boss")
+	robust_searching = TRUE
+	ranged_ignores_vision = TRUE
+	ranged = 1
+	obj_damage = 5
+	vision_range = 6
+	aggro_vision_range = 18
+	environment_smash = ENVIRONMENT_SMASH_NONE  //This is to prevent elites smashing up the mining station, we'll make sure they can smash minerals fine below.
+	harm_intent_damage = 0 //Punching elites gets you nowhere
+	stat_attack = UNCONSCIOUS
+	layer = LARGE_MOB_LAYER
+	sentience_type = SENTIENCE_BOSS
+	hud_type = /datum/hud/lavaland_elite
+	var/chosen_attack = 1
+	var/list/attack_action_types = list()
+	var/can_talk = FALSE
+	var/obj/loot_drop = null
+	
+		
+//Gives player-controlled variants the ability to swap attacks
+/mob/living/simple_animal/hostile/asteroid/elite/Initialize(mapload)
+	. = ..()
+	for(var/action_type in attack_action_types)
+		var/datum/action/innate/elite_attack/attack_action = new action_type()
+		attack_action.Grant(src)
+
+//Prevents elites from attacking members of their faction (can't hurt themselves either) and lets them mine rock with an attack despite not being able to smash walls.
+/mob/living/simple_animal/hostile/asteroid/elite/AttackingTarget()
+	if(istype(target, /mob/living/simple_animal/hostile))
+		var/mob/living/simple_animal/hostile/M = target
+		if(faction_check_mob(M))
+			return FALSE
+	. = ..()
+	if(ismineralturf(target))
```

Comment:
I believe this should be `if(. && ismineralturf(target))`

Reply:
Tested this, turns out it does not work, and this setup is required.